### PR TITLE
support wildcards pattern with common suffix in imports/exports field

### PR DESCRIPTION
**Pull request by bvanjoi**
solve https://github.com/webpack/enhanced-resolve/issues/352

### CODE REVIEW

File: lib/util/entrypoints.js [hunk 1]
```diff
@@ -430,7 +430,24 @@ function targetMapping(
...
 	}
 	assert(mappingTarget, false);
-	return mappingTarget.replace(/\*/g, remainingRequest.replace(/\$/g, "$$"));
+	const wildcardIndex = mappingTarget.indexOf("*");
+
+	if (wildcardIndex !== -1) {
+		const maybeCommonSuffix = mappingTarget.slice(wildcardIndex + 1);
+		if (remainingRequest.endsWith(maybeCommonSuffix)) {
+			return mappingTarget.replace(
+				new RegExp(`\\*${maybeCommonSuffix}`, "g"),
```
Comment (dummdidumm) [inline lib/util/entrypoints.js:439]: I think this needs to escape the characters that could have special meaning to the regex (like `.`)

**Review by dummdidumm — COMMENTED** [review]
An alternative (which would require more work, or more significant changes, probably) would be to use [resolve.exports](https://www.npmjs.com/package/resolve.exports) which aims to consolidate resolution algorithms across tooling for this case.

File: lib/util/entrypoints.js [hunk 2]
```diff
@@ -549,9 +566,10 @@ function walkPath(root, path, target) {
 		node.folder = target;
 	} else {
 		const file = lastNonSlashIndex > 0 ? path.slice(lastNonSlashIndex) : path;
-		if (file.endsWith("*")) {
+		const wildcardsIndex = file.indexOf("*");
```
Comment (alexander-akait) [inline lib/util/entrypoints.js:569]: Weak check, in some filesystem file can have `*` character in the filename, let's use more strick checks, i.e. `*` or `*.js`

File: lib/util/entrypoints.js (hunk 1 above)
Comment (alexander-akait) [reply lib/util/entrypoints.js:439]: There is potential problem, we should handle only `*.js`, not `*` characters

**Review by alexander-akait — CHANGES_REQUESTED** [review]
Let's use more stricky checks

**Comment by alexander-akait** [comment]
And sorry for delay, @bvanjoi do you want to finish this PR
@dummdidumm you can resend this PR if we won't get an answer from @bvanjoi becasei it is so old :disappointed: sorry

**Comment by bvanjoi** [comment]
> And sorry for delay, @bvanjoi do you want to finish this PR

I can refine this pr this weekend (forgive me for being a bit busy at work)

**Comment by alexander-akait** [comment]
@bvanjoi Don't worry feel free to ping me

**Comment by alexander-akait** [comment]
Almost finished, need to more test cases - WIP

**Review by alexander-akait — APPROVED** [review]

**Review by TheLarkInn — APPROVED** [review]

**Pull request merged**